src/main/app/case/to-api-format.ts

```diff
@@ -39,6 +40,104 @@ const fields = {
+        JurisdictionRespondentResidence: data.partnersLifeBasedInEnglandAndWales,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionRespondentResidence: data.partnersLifeBasedInEnglandAndWales,
+      };
+    }
+  },
+  yourDomicileInEnglandWales: (data: Case) => {
+    const connection = addConnection(data);
+    if (connection) {
+      return {
+        JurisdictionConnections: [addConnection(data)],
+        JurisdictionPetitionerDomicile: data.yourDomicileInEnglandWales,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionPetitionerDomicile: data.yourDomicileInEnglandWales,
+      };
+    }
+  },
+  partnersDomicileInEnglandWales: (data: Case) => {
+    const connection = addConnection(data);
+    if (connection) {
+      return {
+        JurisdictionConnections: [addConnection(data)],
+        JurisdictionRespondentDomicile: data.partnersDomicileInEnglandWales,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionRespondentDomicile: data.partnersDomicileInEnglandWales,
+      };
+    }
+  },
+  lastHabituallyResident: (data: Case) => {
+    const connection = addConnection(data);
+    if (connection) {
+      return {
+        JurisdictionConnections: [addConnection(data)],
+        JurisdictionBothLastHabituallyResident: data.lastHabituallyResident,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionBothLastHabituallyResident: data.lastHabituallyResident,
+      };
+    }
+  },
+  livingInEnglandWalesTwelveMonths: (data: Case) => {
+    const connection = addConnection(data);
+    if (connection) {
+      return {
+        JurisdictionConnections: [addConnection(data)],
+        JurisdictionPetHabituallyResLastTwelveMonths: data.livingInEnglandWalesTwelveMonths,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionPetHabituallyResLastTwelveMonths: data.livingInEnglandWalesTwelveMonths,
+      };
+    }
+  },
+  livingInEnglandWalesSixMonths: (data: Case) => {
+    const connection = addConnection(data);
+    if (connection) {
+      return {
+        JurisdictionConnections: [addConnection(data)],
+        JurisdictionPetHabituallyResLastSixMonths: data.livingInEnglandWalesSixMonths,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionPetHabituallyResLastSixMonths: data.livingInEnglandWalesSixMonths,
+      };
+    }
+  },
+  jurisdictionResidualEligible: (data: Case) => {
+    const connection = addConnection(data);
+    if (connection) {
+      return {
+        JurisdictionConnections: [addConnection(data)],
+        JurisdictionResidualEligible: data.jurisdictionResidualEligible,
+      };
+    } else {
+      return {
+        JurisdictionConnections: [],
+        JurisdictionResidualEligible: data.jurisdictionResidualEligible,
+      };
+    }
```

Comment:
all these functions follow the same pattern, we can simplify them all to

```
jurisdictionResidualEligible: (data: Case) => ({
  JurisdictionConnections: getConnections(data),
  JurisdictionResidualEligible: data.jurisdictionResidualEligible,
})
```

Also, why is JurisdictionConnections an array when it's only ever empty or one item?

Reply:
JurisdictionConnections is an array because it's going be used on the checkbox page so needs to be able to hold more than one connection.

And I tried to clean the functions up like that but was getting an error for trying to submit a null value into an array

Reply:
My bad - have just tried cleaning up the functions and it works !